bundles/org.openhab.binding.mqtt.homeassistant/src/test/java/org/openhab/binding/mqtt/homeassistant/internal/component/NumberTests.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mqtt.homeassistant.internal.component;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.junit.jupiter.api.Test;
+import org.openhab.binding.mqtt.generic.values.NumberValue;
+import org.openhab.core.library.types.DecimalType;
+
+/**
+ * Tests for {@link Number}
+ *
+ * @author Cody Cutrer - Initial contribution
+ */
+@NonNullByDefault
+public class NumberTests extends AbstractComponentTests {
+    public static final String CONFIG_TOPIC = "number/0x0000000000000000_number_zigbee2mqtt";
+
+    @SuppressWarnings("null")
+    @Test
+    public void test() throws InterruptedException {
+        var component = discoverComponent(configTopicToMqtt(CONFIG_TOPIC), """
+                    {
+                        "name": "BWA Link Hot Tub Pump 1",
+                        "availability_topic": "homie/bwa/$state",
```

Comment:
Is it "logic" to use homie while this is a test for home assistant ?

Reply:
Heh. Yeah, seems a bit odd at first glance. The test comes from a real life device: https://github.com/ccutrer/balboa_worldwide_app. It supports _both_ Homie and Home Assistant. The latter works by just pointing its configuration to the topics from the former. It's the only Home Assistant-compatible device I currently have that has any Number entities. Lots of people use this device with actual Home Assistant though (because they made me begrudgingly add support for it 😉

Reply:
Ok, thank you for your answer.